clients/src/test/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImplTest.java

```diff
@@ -306,32 +344,49 @@ public void testCompleteQuietly() {
 
     @Test
     public void testSubscribeGeneratesEvent() {
-        consumer = newConsumer();
+        SubscriptionState subscriptions = new SubscriptionState(new LogContext(), OffsetResetStrategy.NONE);
+        consumer = newConsumer(
+            mock(ShareFetchBuffer.class),
+            subscriptions,
+            "group-id",
+            "client-id");
         String topic = "topic1";
-        consumer.subscribe(singletonList(topic));
+        final List<String> subscriptionTopic = singletonList(topic);
+        completeShareSubscriptionChangeApplicationEventSuccessfully(subscriptions, subscriptionTopic);
+        consumer.subscribe(subscriptionTopic);
         assertEquals(singleton(topic), consumer.subscription());
-        verify(applicationEventHandler).add(ArgumentMatchers.isA(ShareSubscriptionChangeEvent.class));
+        verify(applicationEventHandler).addAndGet(ArgumentMatchers.isA(ShareSubscriptionChangeEvent.class));
     }
 
     @Test
     public void testUnsubscribeGeneratesUnsubscribeEvent() {
-        consumer = newConsumer();
-        completeShareUnsubscribeApplicationEventSuccessfully();
+        SubscriptionState subscriptions = new SubscriptionState(new LogContext(), OffsetResetStrategy.NONE);
+        consumer = newConsumer(
+            mock(ShareFetchBuffer.class),
+            subscriptions,
+            "group-id",
+            "client-id");
+        completeShareUnsubscribeApplicationEventSuccessfully(subscriptions);
 
         consumer.unsubscribe();
 
         assertTrue(consumer.subscription().isEmpty());
```

Comment:
this assertion doesn't bring much value now right? The subscription is cleared in the background now, this is only passing here because we mock it ourselves in the `completeShareUnsubscribeApplicationEventSuccessfully` (verifying that the event was generated is all that matters I would say)